test/test_pipeline_yaml.py

```diff
@@ -792,3 +852,46 @@ def test_save_yaml_overwrite(tmp_path):
     with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
         content = saved_yaml.read()
         assert content != ""
+
+
+def test_save_yaml_with_custom_node_with_params(caplog, tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    pipeline.add_node(CustomNode(param=10), name="custom_node", inputs=["Query"])
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
+
+    loaded_pipeline = Pipeline.load_from_yaml(tmp_path / "saved_pipeline.yml")
+
+    original_node = pipeline.get_config()["components"][0]
+    loaded_node = loaded_pipeline.get_config()["components"][0]
+
+    assert original_node == loaded_node
+
+
+def test_save_yaml_with_custom_node_with_positional_params(caplog, tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int = 1):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    with caplog.at_level(logging.WARNING):
+        pipeline.add_node(CustomNode(10), name="custom_node", inputs=["Query"])
+        assert (
+            "Unnamed __init__ parameters will not be saved to YAML "
+            "if Pipeline.save_to_yaml() is called" in caplog.text
+        )
+
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
+
+    loaded_pipeline = Pipeline.load_from_yaml(tmp_path / "saved_pipeline.yml")
+
+    original_node = pipeline.get_config()["components"][0]
+    loaded_node = loaded_pipeline.get_config()["components"][0]
+
+    assert original_node["params"]["param"] == 10
+    assert loaded_node["params"]["param"] == 1
```

Comment:
Here param should not be present in the config at all as it is the default value.